uploadNeuroDB/minc_deletion.pl

```diff
@@ -0,0 +1,322 @@
+    }
+    print "\n" . $query;
+  }
+}
+
+
+# Delete from FS:
+# get the file names
+$query = "select f.File, f.TarchiveSource, f.SessionID, pf.`VALUE` from files as f ".
+         "left join parameter_file as pf using (FileID) where ".
+         "pf.ParameterTypeID = (select pt.ParameterTypeID from parameter_type as pt where pt.Name = 'check_pic_filename') and ".
+         "pf.FileID IN ";
+
+# Useful database tracing for troubleshooting
+# $dbh->trace(5);
+
+if ($seriesuid) {
+  $query .= "(SELECT FileID FROM files WHERE SeriesUID = ?)";
+  $sth = $dbh->prepare($query);
+  $rvl = $sth->execute($seriesuid);
+} elsif ($fileid) {
+  $query .= "?";
+  $sth = $dbh->prepare($query);
+  $rvl = $sth->execute($fileid);
+}
+
+if ($sth->err) {
+  die "ERROR! return code:" . $sth->err . " error msg: " . $sth->errstr . "\n";
+}
+
+if (defined $rvl && $rvl == 0) {
+  die "Can't rename if there is no value return from: \n" . $query . "\n";
+}
+
+
+my ($tarchiveid, $sessionid, @pic_path, $jiv_header, $jiv_raw_byte, $file, $dir, $ext, $nii_file, @candid);
+
+while (my $f = $sth->fetchrow_hashref()) {
+  $tarchiveid   = $f->{'TarchiveSource'};
+  $sessionid    = $f->{'SessionID'};
+  @pic_path     = split /_check/, $f->{'VALUE'};
+  $jiv_header   = $pic_path[0] . ".header";
+  $jiv_raw_byte = $pic_path[0] . ".raw_byte.gz";
+  ($file, $dir, $ext) = fileparse($f->{'File'});
+  $nii_file     = basename($file, ".mnc") . ".nii";
+  @candid = split("/", $dir);
+
+  if ($ARGV[0] eq "confirm") {
+    # Let's make directories
+    make_path($data_dir . "/archive/"     . $dir) unless(-d  $data_dir . "/archive/"     . $dir);
+    make_path($data_dir . "/archive/pic/" . $candid[1]) unless(-d  $data_dir . "/archive/pic/" . $candid[1]);
+    make_path($data_dir . "/archive/jiv/" . $candid[1]) unless(-d  $data_dir . "/archive/jiv/" . $candid[1]);
+
+    if (-e $data_dir . "/" . $dir . $nii_file) {
+      rename($data_dir . "/" . $dir . $nii_file, $data_dir . "/archive/" . $dir . $nii_file);
+    }
+    rename($data_dir . "/"     . $f->{'File'}, $data_dir . "/archive/" . $f->{'File'});
+    rename($data_dir . "/pic/" . $f->{'VALUE'}, $data_dir . "/archive/pic/" . $f->{'VALUE'});
+    rename($data_dir . "/jiv/" . $jiv_header, $data_dir . "/archive/jiv/" . $jiv_header);
+    rename($data_dir . "/jiv/" . $jiv_raw_byte, $data_dir . "/archive/jiv/" . $jiv_raw_byte);
+    print "\nMoving these files to archive:\n";
+  } else {
+    print "\nFiles that will be moved when rerunning the script using the confirm option:\n";
+  }
+
+  print $data_dir . "/"     . $f->{'File'} . "\n";
+  if (-e $data_dir . "/"    . $dir . $nii_file) {
+    print $data_dir . "/"   . $dir . $nii_file . "\n";
+  }
+  print $data_dir . "/pic/" . $f->{'VALUE'} . "\n";
+  print $data_dir . "/jiv/" . $jiv_header . "\n";
+  print $data_dir . "/jiv/" . $jiv_raw_byte . "\n";
+}
+
+print "\nDelete from DB";
+# Delete from DB
+selORdel("parameter_file","Value");
+selORdel("files_qcstatus","QCStatus");
```

Comment:
look at the delete of lines 238 and 239;
before deleting from these two tables; you should have a user input to confirm this is to be deleted
(can map the information and then when re-inserting the new file, can use SeriesUID and EchoTime to re-insert the QC information for the newly inserted FileID)